.github/workflows/build_docs.yml

```
@@ -0,0 +1,48 @@
+name: Publish docs to github.io
+
+on:
+  push:
+    branches:
+      - main
+      - docs/*
+
+jobs:
+  publish-docs:
+    name: Publish developer docs
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Setup just
+        uses: extractions/setup-just@v1
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+
+      - name: Generate docs
+        run: |
+            just install
+            just make-docs
+            mv docs/_build/html /tmp/gh-pages
+
+      - name: Check out gh-pages branch
+        uses: actions/checkout@v3
+        with:
+          ref: gh-pages
+
+      - name: Deploy docs
+        uses: peaceiris/actions-gh-pages@v3.7.3
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_dir: /tmp/gh-pages
+          force_orphan: true
+
+      - name: Checkout branch again to enable cleaning
+        if: always()
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.ref_name }}
+
```

With the `peaceiris/actions-gh-pages` action you actually don't need to check out the `gh-pages` or anything yourself, the action handles that for you.

You can even skip the `mv` step in the preceding script after building the docs as the action handles all that for you as well.

```suggestion
      - name: Generate docs
        run: |
            just install
            just make-docs

      - name: Deploy docs
        uses: peaceiris/actions-gh-pages@v3.7.3
        with:
          github_token: ${{ secrets.GITHUB_TOKEN }}
          publish_dir: ./docs/_build/html
          force_orphan: true
```

---

Oh awesome! This was something I took from WordPress/openverse-api, so I guess we could go change it there too 😄 : 

https://github.com/WordPress/openverse-api/blob/a44f8d19cc3b2612469bbf20a3aac54c44086712/.github/workflows/ci_cd.yml#L310-L358